analyzers/src/SonarAnalyzer.Common/Helpers/SonarLintXmlReader.cs

```diff
@@ -0,0 +1,114 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2023 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.IO;
+using System.Text;
+using System.Xml;
+using System.Xml.Serialization;
+using Microsoft.CodeAnalysis.Text;
+
+namespace SonarAnalyzer.Helpers;
+
+public class SonarLintXmlReader
+{
+    public static readonly SonarLintXmlReader Empty = new(null, LanguageNames.CSharp);
+
+    private readonly SonarLintXml sonarLintXml;
+    private readonly string propertyLanguage;
+
+    private SonarLintXmlSettingsReader settings;
+    public SonarLintXmlSettingsReader Settings => settings ??= new(sonarLintXml, propertyLanguage);
+
+    public SonarLintXmlReader(SourceText originalXml, string language)
+    {
+        sonarLintXml = originalXml == null ? SonarLintXml.Empty : ParseContent(originalXml);
+        propertyLanguage = language;
+    }
+
+    private static SonarLintXml ParseContent(SourceText sonarProjectConfig)
+    {
+        try
+        {
+            var serializer = new XmlSerializer(typeof(SonarLintXml));
+            var byteArray = Encoding.UTF8.GetBytes(sonarProjectConfig.ToString());
+            var stream = new MemoryStream(byteArray);
+            using var sr = new StreamReader(stream, Encoding.UTF8, false);
+            using var reader = XmlReader.Create(sr);
+            return (SonarLintXml)serializer.Deserialize(reader);
+        }
+        catch
+        {
+            return SonarLintXml.Empty;
+        }
+    }
+}
+
+public class SonarLintXmlSettingsReader
```

Comment:
Why did you introduce this additional level of indirection? Why didn`t you put the settings directly on the reader?

Reply:
IMO it is more readable this way. 
Instead of bunching all the settings (and the rule parameters too in the future) on the same class, having the base class that is specifically taking care of the parsing and two classes that deal with properties inside each block seems more approachable to me. Does this level of indirection still seem unnecessary to you?

Reply:
To a point, I understand you want to separate the concerns of the two classes and at the same time, I think it is an overkill, this way the user will have to write `.Settings.Parameter` at every usage. I would move the properties to the reader itself. (BTW the reading of the parameters is out of scope in this PR it should have been part of the other PRs)

Reply:
Ok I will merge it into the reader then. Yes, I'm aware. Unfortunately I realized only after the implementation there were specific tickets for each property. Let's leave it as is, since with this PR we are simply reading them; the PR about each property will change the functionality to use this value instead of the old one. Another possibility is to remove them from here and place the reading on each specific PR too, WDYT?

Reply:
Of course, leave it here, it does not make sense to remove them.